src/czbenchmarks/cli/cli_run.py

```diff
@@ -56,81 +49,196 @@ def run(ctx):
+
+            # 5. Run the task(s)
+            all_results = []
+
+            # Run baseline if requested
+            if compute_baseline:
+                log.info("--- Running Baseline ---")
+                # For baseline, the input is always the raw expression data
+                raw_data = [d.adata.X for d in datasets]
+                baseline_input = raw_data if len(raw_data) > 1 else raw_data[0]
+
+                baseline_results = run_task(
+                    task_name=invoked_task_name,
+                    cell_representation=baseline_input,
+                    run_baseline=True,
+                    baseline_params=baseline_args,
+                    task_params=processed_task_args,
+                    random_seed=random_seed,
+                )
+                all_results.extend(baseline_results)
+                log.info("--- Baseline Run Complete ---")
+
+            # Run with model embedding
+            log.info("--- Running Model ---")
+            model_embeddings = [load_from_path(p) for p in model_embedding_paths]
+            model_input = (
+                model_embeddings if len(model_embeddings) > 1 else model_embeddings[0]
             )
-            log.info("Running task on computed baseline embedding...")
-            baseline_metric_results = task_instance.run(
-                cell_representation=baseline_embedding,
-                task_input=task_input,
+
+            model_results = run_task(
+                task_name=invoked_task_name,
+                cell_representation=model_input,
+                run_baseline=False,  # Never compute baseline on an embedding
+                task_params=processed_task_args,
+                random_seed=random_seed,
             )
-            all_metric_results.extend(baseline_metric_results)
-            log.info(
-                f"Computed {len(baseline_metric_results)} metric(s) for the baseline."
+            all_results.extend(model_results)
+            log.info("--- Model Run Complete ---")
+
+            # 6. Write results
+            metric_objects = [MetricResult(**res) for res in all_results]
+            write_results(metric_objects, output_file)
+            click.echo(f"✅ Task '{invoked_task_name}' completed successfully.")
+
+        except (click.ClickException, ValueError) as e:
+            log.error(f"Task '{invoked_task_name}' failed: {e}")
+            raise click.ClickException(str(e))  # Re-raise as a clean CLI error
+        except Exception as e:
+            log.error(
+                f"An unexpected error occurred in task '{invoked_task_name}': {e}",
+                exc_info=True,
             )
-
-        write_results(all_metric_results, output_file)
-        click.echo(f"✅ Task '{invoked_task_name}' completed successfully.")
+            raise click.ClickException(f"An unexpected error occurred: {e}")
+
+    # --- Dynamically Add Options to the Command ---
+    def add_click_options(options):
+        def decorator(f):
```

Comment:
should be
```
@functools.wraps(f)
def decorator(f)
```
That way the decorated function still retains its original name and doctring and stuff